lucene/core/src/java/org/apache/lucene/search/TopFieldCollector.java

```diff
@@ -68,6 +68,20 @@ public void setScorer(Scorable scorer) throws IOException {
   }
 
   static boolean canEarlyTerminate(Sort searchSort, Sort indexSort) {
+    return canEarlyTerminateOnDocId(searchSort, indexSort) ||
+           canEarlyTerminateOnPrefix(searchSort, indexSort);
+  }
+
+  private static boolean canEarlyTerminateOnDocId(Sort searchSort, Sort indexSort) {
+    final SortField[] fields1 = searchSort.getSort();
+    final SortField[] fields2 = indexSort.getSort();
+    return fields1.length == 1 &&
```

Comment:
It'd be weird, but we can also safely early terminate even if there are other sort fields after `docid` right (since `docid` is a total sort)?  So we don't need to insist the length is 1?

Reply:
Oh, yes, we don't need it, I'll remove these conditions.

Reply:
Fixed in 7ef9b96746503ca0672d9e65b41555bc4ae567b3.